presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -749,7 +753,25 @@ public void testQueryLoggingCount()
                             ImmutableList.of()),
                     new Duration(1, MINUTES));
 
-            long beforeCompletedQueriesCount = queryManager.getStats().getCompletedQueries().getTotalCount();
+            // The use of waitUntilStable below is because of intermittent failures of this test.
```

Comment:
This comment is confusing, because I understand from this comment that it's *not* safe to get the value of `beforeSubmittedQueriesCount ` immediately after queries finish, however we don't use `waitUntilStable` for it below like we do for the other counter (in other words, if it's not safe why don't we wrap it with `waitUntilStable`). In fact, when I look at the code I think it's safe to get the number of submitted queries as that counter is updated synchronously (from `createQuery()` -> `addStatsListeners()` -- not in a final query info listener unlike the other counter).

How about we add this comment instead?

```
// We cannot simply get the number of completed queries as soon as all the queries are completed, because this counter may not be up-to-date at that point.
// The completed queries counter is updated in a final query info listener, which is called eventually.
// Therefore, here we wait until the value of this counter gets stable.
```

Reply:
Oh good catch! I looked at just one counter and wrongly generalised how it is set it to both. Yet the reason why it is not wrapped with `waitUntilStable` is that I assumed that when one stabilises, then the other should be good too. 

Anyway, whole logic is wrong for other counter so I will change the comment and commit message.